From 21fe4ddda1f36a4d2419ac712633cb1caef879eb Mon Sep 17 00:00:00 2001
From: Sivasubramaniyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
Date: Thu, 25 Apr 2024 17:06:39 +0530
Subject: [PATCH 1/4] ADSD3500 user class private control ID address

Signed-off-by: Sivasubramaniyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
---
 include/uapi/linux/v4l2-controls.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/uapi/linux/v4l2-controls.h b/include/uapi/linux/v4l2-controls.h
index 99c3f5e99da7..dea036f4bbb2 100644
--- a/include/uapi/linux/v4l2-controls.h
+++ b/include/uapi/linux/v4l2-controls.h
@@ -169,6 +169,12 @@ enum v4l2_colorfx {
  */
 #define V4L2_CID_USER_ATMEL_ISC_BASE		(V4L2_CID_USER_BASE + 0x10c0)
 
+/*
+ *  *  *  * * The base for the addicmos driver controls.
+ *   *   *   * * We reserve 16 controls for this driver.
+ *    *    *    * */
+#define V4L2_CID_USER_ADITOF_BASE         (V4L2_CID_USER_BASE + 0x10d0)
+
 /*
  * The base for the CODA driver controls.
  * We reserve 16 controls for this driver.
-- 
2.25.1

